legacy/python/platform/src/actions/setup_tool.rs

```diff
@@ -0,0 +1,50 @@
+use moon_action::Operation;
+use moon_python_tool::PythonTool;
+use starbase_utils::fs;
+use std::path::Path;
+
+pub async fn setup_tool(python: &PythonTool, workspace_root: &Path) -> miette::Result<()> {
+    let mut operations = vec![];
+
+    if let Some(pip_config) = &python.config.pip {
+        let virtual_environment = &workspace_root.join(python.config.venv_name.clone());
+
+        if !virtual_environment.exists() {
```

Comment:
Is venv coupled with a requirements.txt? Or are they separate?

If venv stuff can be per-project, this venv logic should be in the sync_project action, since the setup_tool doesn't run in the project directory.

Reply:
As mentioned below, strategy changed and now working with single version policy and project specific configurations.